detectron2/modeling/poolers.py

```diff
@@ -59,8 +59,12 @@ def assign_boxes_to_levels(
 
 
 def _fmt_box_list(box_tensor, batch_index: int):
-    repeated_index = torch.full(
-        (len(box_tensor), 1), batch_index, dtype=box_tensor.dtype, device=box_tensor.device
+    repeated_index = torch.full_like(
+        box_tensor[:, :1],
+        batch_index,
+        dtype=box_tensor.dtype,
+        device=box_tensor.device,
+        layout=torch.strided,
```

Comment:
layout= is probably not needed

Reply:
Indeed it probably isn't. I don't know the probability that a sparse tensor of bounding boxes can be passed in. If this is zero the `layout=torch.strided` is not necessary. I don't think having it there will hurt either.

Reply:
`box_tensor[:, :1]` will fail on sparse tensor, if it ever appears

Reply:
You're right. I'll remove it then.